### ISSUE TITLE
This isn’t compiling with nginx 1.9.11

### ISSUE THREAD

**Issue by jonnybarnes** [google/ngx_brotli#16]
The end of the output for `make`:
```
	-o objs/addon/src/ngx_http_brotli_filter_module.o \
	/home/jonny/server-files/nginx/ngx_brotli/src/ngx_http_brotli_filter_module.cc
/home/jonny/server-files/nginx/ngx_brotli/src/ngx_http_brotli_filter_module.cc:173:1: error: deprecated conversion from string constant to ‘char*’ [-Werror=write-strings]
 };
 ^
cc1plus: all warnings being treated as errors
objs/Makefile:1244: recipe for target 'objs/addon/src/ngx_http_brotli_filter_module.o' failed
make[1]: *** [objs/addon/src/ngx_http_brotli_filter_module.o] Error 1
make[1]: Leaving directory '/home/jonny/server-files/nginx/nginx-1.9.11'
Makefile:8: recipe for target 'build' failed
make: *** [build] Error 2
```

**Comment by charlesportwoodii** [comment]
Hi,

I'm seeing the same issue for Nginx 1.9.11 with the following configure line:

```
./configure 
--with-http_geoip_module 
--with-http_realip_module 
--with-http_ssl_module 
--with-http_gunzip_module 
--with-http_addition_module 
--with-http_auth_request_module 
--with-http_gzip_static_module 
--with-http_stub_status_module 
--with-http_v2_module 
--with-http_sub_module 
--with-ipv6
--prefix=/etc/nginx
--sbin-path=/usr/bin/nginx
--error-log-path=/var/log/nginx/error.log
--pid-path=/var/run/nginx.pid
--http-log-path=/var/log/nginx/access.log
--add-module=modules/ngx_brotli
```

Configure passes, but ```ngx_brotli``` spits out the following extra information. The output looks somewhat similar to that which is outputted by ```ngx_pagespeed``` https://github.com/pagespeed/ngx_pagespeed/issues/1110

```
adding module in modules/ngx_brotli
checking for Brotli library ... found
sed: -e expression #1, char 0: no previous regular expression
 + ngx_brotli was configured
```

Make then errors out with the following:

```
cc -c -pipe  -O -W -Wall -Wpointer-arith -Wno-unused-parameter -Werror -g   -I src/core -I src/event -I src/event/modules -I src/os/unix -I objs -I src/http -I src/http/modules -I src/http/v2 -I src/mail \
		-o objs/addon/src/ngx_http_brotli_filter_module.o \
		modules/ngx_brotli/src/ngx_http_brotli_filter_module.cc
modules/ngx_brotli/src/ngx_http_brotli_filter_module.cc:173:1: error: deprecated conversion from string constant to ‘char*’ [-Werror=write-strings]
 };
 ^
cc1plus: all warnings being treated as errors
make[1]: *** [objs/addon/src/ngx_http_brotli_filter_module.o] Error 1
make[1]: Leaving directory `/tmp/nginx-1.9.11'
make: *** [build] Error 2
```

**Comment by jonnybarnes** [comment]
I got the same `sed` line as well.

**Comment by charlesportwoodii** [comment]
It looks like if you configure ```ngx_brotli``` with ```export NGX_BROTLI_STATIC_MODULE_ONLY=1```, compilation proceeds without error.

```
export NGX_BROTLI_STATIC_MODULE_ONLY=1 && \
./configure --add-module=modules/ngx_brotli
```

**Comment by centminmod** [comment]
@charlesportwoodii tried that but get 
```
Starting nginx: nginx: [emerg] unknown directive "brotli" 
```

despite the ngx_brotli module showing up at
```
--with-http_gzip_static_module --add-module=../ngx_brotli
```
within
```
nginx -V
nginx version: nginx/1.9.11
built by clang 3.4.2 (tags/RELEASE_34/dot2-final)
built with LibreSSL 2.2.6
TLS SNI support enabled
configure arguments: --with-ld-opt='-lrt -ljemalloc -Wl,-z,relro -Wl,-rpath,/usr/local/lib' --with-cc-opt='-m64 -mtune=native -mfpmath=sse -g -O2 -fstack-protector --param=ssp-buffer-size=4 -Wformat -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wno-sign-compare -Wno-string-plus-int -Wno-deprecated-declarations -Wno-unused-parameter -Wno-unused-const-variable -Wno-conditional-uninitialized -Wno-mismatched-tags -Wno-c++11-extensions -Wno-sometimes-uninitialized -Wno-parentheses-equality -Wno-tautological-compare -Wno-self-assign -Wno-deprecated-register -Wno-deprecated -Wno-invalid-source-encoding -Wno-pointer-sign -Wno-parentheses -Wno-enum-conversion -Wno-c++11-compat-deprecated-writable-strings' --sbin-path=/usr/local/sbin/nginx --conf-path=/usr/local/nginx/conf/nginx.conf --with-http_stub_status_module --with-http_secure_link_module --with-openssl-opt=enable-tlsext --add-module=../nginx-module-vts --with-libatomic --with-threads --with-stream --with-stream_ssl_module --with-http_gzip_static_module --add-module=../ngx_brotli --with-http_sub_module --with-http_addition_module --with-http_image_filter_module --with-http_geoip_module --with-http_realip_module --add-module=../nginx-accesskey-2.0.3 --add-module=../nginx-http-concat-master --add-module=../ngx-fancyindex-ngx-fancyindex --add-module=../ngx_cache_purge-2.3 --add-module=../ngx_devel_kit-0.2.19 --add-module=../set-misc-nginx-module-0.29 --add-module=../echo-nginx-module-0.58 --add-module=../redis2-nginx-module-0.12 --add-module=../ngx_http_redis-0.3.7 --add-module=../nginx_upstream_check_module-0.3.0 --add-module=../openresty-memc-nginx-module-4f6f78f --add-module=../openresty-srcache-nginx-module-ffa9ab7 --add-module=../headers-more-nginx-module-0.29 --with-pcre=../pcre-8.38 --with-pcre-jit --with-http_ssl_module --with-http_v2_module --with-openssl=../libressl-2.2.6
```

**Comment by PiotrSikora** [comment]
@centminmod : using `NGX_BROTLI_STATIC_MODULE_ONLY=1` provides only `brotli_static` module (i.e. serving of pre-compressed files).

This is being worked on, expect fixes later today.

**Comment by centminmod** [comment]
ah thanks @PiotrSikora for the clarification ..appreciate your work on this :)

**Comment by PiotrSikora** [comment]
Fixed in d719f61e1793e87ec46dabfec87439df7256f281.

**Comment by centminmod** [comment]
haha nice just noticed the fix when i recompiled nginx just seconds ago https://community.centminmod.com/posts/26127/ !

thanks @PiotrSikora
